File: examples/basic.py
#!/usr/bin/env python3
"""
Basic example with OpenAI SDK
"""

from openai import OpenAI
import requests
import os

BASE_URL = "http://localhost:3000"
API_KEY_FILE = ".example-api-key"

print("=" * 50)
print("DeepSeek API - OpenAI SDK Example")
print("=" * 50)

# Step 1: Get API key via register endpoint
print("\n1. Getting API key...")
response = requests.post(
    f"{BASE_URL}/v1/register",
    json={"email": "your@email.com", "password": "yourpassword"}
)

if response.status_code != 200:
    print(f"Error: {response.status_code}")
    print(response.text)
    exit(1)

api_key = response.json()["api_key"]
print(f"✓ API key: {api_key}")

# Save for future use
with open(API_KEY_FILE, "w") as f:
    f.write(api_key)

# Step 2: Use OpenAI SDK
print("\n2. Sending request with OpenAI SDK...")
client = OpenAI(
    base_url=f"{BASE_URL}/v1",
    api_key=api_key
)

response = client.chat.completions.create(
    model="deepseek-chat",
    messages=[{"role": "user", "content": "Hello! Tell me a short joke"}]
)

print(f"✓ Response: {response.choices[0].message.content}")

# Step 3: With tool calling
print("\n3. Tool calling with OpenAI SDK...")
tools = [{
    "type": "function",
    "function": {
        "name": "get_weather",
        "description": "Get current weather",
        "parameters": {
            "type": "object",
            "properties": {"location": {"type": "string"}},
            "required": ["location"]
        }
    }
}]

response = client.chat.completions.create(
    model="deepseek-chat",
    messages=[{"role": "user", "content": "What's the weather in London?"}],
    tools=tools
)

message = response.choices[0].message
if message.tool_calls:
    print("✓ Tool call detected:")
    for tool in message.tool_calls:
        print(f"   {tool.function.name}({tool.function.arguments})")
else:
    print(f"✓ Response: {message.content}")

print("\n" + "=" * 50)
print("OpenAI SDK example completed!")
print("=" * 50)

File: examples/basic.js
#!/usr/bin/env node

/**
 * Basic example - first request with auto-login and subsequent requests with API key
 */

const BASE_URL = "http://localhost:3000";
const fs = require('fs');
const path = require('path');

const API_KEY_FILE = path.join(__dirname, '.example-api-key');

async function main() {
    console.log("=".repeat(50));
    console.log("DeepSeek API - Basic Example");
    console.log("=".repeat(50));

    let apiKey = null;

    // Try to load existing API key
    if (fs.existsSync(API_KEY_FILE)) {
        apiKey = fs.readFileSync(API_KEY_FILE, 'utf-8').trim();
        console.log(`\n✓ Loaded existing API key: ${apiKey}`);
    }

    // Step 1: First request with email/password (if no API key)
    if (!apiKey) {
        console.log("\n1. Sending first request with email/password...");
        
        const response = await fetch(`${BASE_URL}/v1/chat/completions`, {
            method: "POST",
            headers: { "Content-Type": "application/json" },
            body: JSON.stringify({
                email: "your@email.com",      // Replace with your DeepSeek email
                password: "yourpassword",      // Replace with your DeepSeek password
                messages: [{ role: "user", content: "Say 'Hello, API!'" }]
            })
        });

        if (!response.ok) {
            console.error(`Error: ${response.status}`);
            const error = await response.text();
            console.error(error);
            process.exit(1);
        }

        // Get API key from response headers
        apiKey = response.headers.get("X-API-Key");
        console.log(`✓ API key received: ${apiKey}`);

        const data = await response.json();
        console.log(`✓ Response: ${data.choices[0].message.content}`);

        // Save API key for future use
        fs.writeFileSync(API_KEY_FILE, apiKey);
        console.log("\n✓ API key saved to .example-api-key");
    }

    // Step 2: Subsequent request with API key
    console.log("\n2. Sending second request with API key...");
    
    const response = await fetch(`${BASE_URL}/v1/chat/completions`, {
        method: "POST",
        headers: {
            "Authorization": `Bearer ${apiKey}`,
            "Content-Type": "application/json"
        },
        body: JSON.stringify({
            messages: [{ role: "user", content: "Tell me a short joke" }]
        })
    });

    const data = await response.json();
    console.log(`✓ Response: ${data.choices[0].message.content}`);

    console.log("\n" + "=".repeat(50));
    console.log("Basic example completed!");
    console.log("=".repeat(50));
}

main().catch(console.error);